build.psm1

```diff
@@ -2023,15 +2023,48 @@ function script:precheck([string]$command, [string]$missedMessage) {
 
 # this function wraps native command Execution
 # for more information, read https://mnaoumov.wordpress.com/2015/01/11/execution-of-external-commands-in-powershell-done-right/
-function script:Start-NativeExecution([scriptblock]$sb, [switch]$IgnoreExitcode)
+function script:Start-NativeExecution
 {
+    param(
+        [scriptblock]$sb,
+        [switch]$IgnoreExitcode,
+        [switch]$VerboseOutputOnError
+    )
     $backupEAP = $script:ErrorActionPreference
     $script:ErrorActionPreference = "Continue"
     try {
-        & $sb
+        if($VerboseOutputOnError.IsPresent)
+        {
+            $output = & $sb
+        }
+        else
+        {
+            & $sb
+        }
+
         # note, if $sb doesn't have a native invocation, $LASTEXITCODE will
         # point to the obsolete value
         if ($LASTEXITCODE -ne 0 -and -not $IgnoreExitcode) {
+            if($VerboseOutputOnError.IsPresent -and $Output)
+            {
+                $output | Out-String | Write-Verbose -Verbose
+            }
+
+            # Get caller location for easier debugging
+            $callerLocation = (Get-PSCallStack -ErrorAction SilentlyContinue)[1].Location
+            if($callerLocation)
```

Comment:
I guess `$callerLocation` will never be empty or null. When there is no location, it will be the string `"<No file>"`, and thus this if check will always be true.

Reply:
the check was more in case the call failed.

Reply:
If `Get-PSCallStack` fails, then the indexing operation `[1]` will result in an exception. For example
```
PS > (gcm abcfa -ErrorAction SilentlyContinue)[0]
Cannot index into a null array.
```

Reply:
Can you please respond to this comment? It seems there is not point to have `if($callerLocation)`.

Reply:
Fixed